src/components/KModal/KModal.vue

```diff
@@ -293,160 +235,110 @@ watch(() => props.isVisible, async (isVisible) => {
-
-  if (props.isVisible) {
-    // Hide body overflow
-    document?.body?.classList.add('k-modal-overflow-hidden')
-  }
 })
 
 onUnmounted(() => {
   document.removeEventListener('keydown', handleKeydown)
-  // Reset body overflow
-  document?.body?.classList.remove('k-modal-overflow-hidden')
 })
 </script>
 
 <style lang="scss" scoped>
-@import '@/styles/tmp-variables';
-@import '@/styles/mixins';
-
-.k-modal-backdrop {
-  background-color: $tmp-color-backdrop;
-  inset: 0;
-  position: fixed;
-  z-index: 1100;
-}
-
-// Allow modal backdrop to scroll if viewport is shorter than modal
-.k-modal-overflow-hidden .k-modal-backdrop {
-  overflow: auto;
-}
-
 .k-modal {
-  // use a var to ensure correct sizing of .k-modal-header-image
-  $k-modal-padding: var(--kui-space-90, $kui-space-90);
-
-  .k-modal-dialog {
-    background: var(--kui-color-background, $kui-color-background);
-    border: initial;
-    border-radius: var(--kui-border-radius-20, $kui-border-radius-20);
-    box-shadow: 0px 0px 12px 0px $tmp-color-black-10;
-    margin: var(--kui-space-110, $kui-space-110) var(--kui-space-auto, $kui-space-auto);
-    overflow: hidden;
-    padding: $k-modal-padding;
-    position: relative;
-    width: auto;
-    z-index: 9999;
-
-    @media (min-width: $kui-breakpoint-phablet) {
-      max-width: v-bind('$props.maxWidth');
-    }
-
-    .close-button {
-      position: absolute;
-      right: 24px;
-      top: 24px;
-      // 1 more than .k-modal-dialog
-      z-index: 10000;
-
-      .k-button {
-        margin-top: calc(-1 * var(--kui-space-40, $kui-space-40));
-        padding: var(--kui-space-40, $kui-space-40);
-        @include non-visual-button;
-      }
-    }
+  .modal-backdrop {
+    align-items: center;
+    background-color: var(--kui-color-background-overlay, $kui-color-background-overlay);
+    display: flex;
+    inset: 0;
+    position: fixed;
+    z-index: 1100;
   }
 
-  .k-modal-content {
-    display: flex;
-    flex-direction: column;
-    position: relative;
-
-    .k-modal-header-image {
-      display: flex !important;
-      margin-bottom: var(--kui-space-90, $kui-space-90);
-      /* stylelint-disable-next-line custom-property-no-missing-var-function */
-      margin-left: calc(#{$k-modal-padding} * -1);
-      /* stylelint-disable-next-line custom-property-no-missing-var-function */
-      margin-right: calc(#{$k-modal-padding} * -1);
-      /* stylelint-disable-next-line custom-property-no-missing-var-function */
-      margin-top: calc(#{$k-modal-padding} * -1);
+  .modal-container {
+    background-color: var(--kui-color-background, $kui-color-background);
+    border: var(--kui-border-width-10, $kui-border-width-10) solid var(--kui-color-border, $kui-color-border);
+    border-radius: var(--kui-border-radius-40, $kui-border-radius-40);
+    box-shadow: var(--kui-shadow, $kui-shadow);
+    margin: var(--kui-space-0, $kui-space-0) var(--kui-space-auto, $kui-space-auto);
+    width: v-bind('widthValue');
```

Comment:
Unsure if this is the right line, but on all viewports (especially mobile) it would be nice to have a little padding to the left and right of the modal white background instead of it going edge to edge
![image](https://github.com/Kong/kongponents/assets/2229946/3cbb4906-1e02-4645-84e5-779c95808ff1)